### PR TITLE
[Windows][melodic] Fix the run_tests build breaks on Windows.

### PR DESCRIPTION
**Pull request by seanyen**
This change is fixing the run_tests build breaks when `catkin_make run_tests` on Windows. `random` doesn't exist on Windows so switch to C++ `<random>` library for better portability.

### CODE REVIEW

File: src/test/send_images.cpp [hunk 1]
```diff
@@ -66,14 +67,16 @@ int main( int argc, char **argv )
     msg.step = width * 3;
 
     int count = 0;
+    std::default_random_engine seed;
```
Comment (rhaschke) [inline src/test/send_images.cpp]: The engine isn't a seed. It's the random number generator.
```suggestion
    std::default_random_engine random_generator;
```

File: src/test/send_images.cpp [hunk 2]
```diff
@@ -66,14 +67,16 @@ int main( int argc, char **argv )
...
     while( ros::ok() )
     {
       for( int x = 0; x < width; x++ )
       {
         for( int y = 0; y < height; y++ )
         {
           int index = (x + y * width) * 3;
-          long int rand = random();
+          std::uniform_int_distribution<int> random(0, RAND_MAX);
```
Comment (rhaschke) [inline src/test/send_images.cpp]: This one maps pseudo random numbers into a random numbers of a specific, here uniform, distribution.
```suggestion
          std::uniform_int_distribution<int> uniform(0, RAND_MAX);
```

File: src/test/send_images.cpp [hunk 3]
```diff
@@ -66,14 +67,16 @@ int main( int argc, char **argv )
...
+          auto rand = random(seed);
```
Comment (rhaschke) [inline src/test/send_images.cpp]:
```suggestion
          auto rand = uniform(random_generator);
```

**Comment by seanyen** [comment]
@rhaschke Thank for the feedbacks! Fixed the naming.

**Pull request merged**